src/plugins/Polls/define.tsx

```diff
@@ -0,0 +1,18 @@
+import React from 'react'
+import type { PluginConfig } from '../plugin'
+import type { PollMetaData } from './types'
+import { readTypedMessageMetadata } from '../../extension/background-script/CryptoServices/utils'
+import PollsInPost from './UI/PollsInPost'
+
+export const PollsPluginDefine: PluginConfig = {
+    pluginName: 'Poll',
+    identifier: 'com.maskbook.poll',
+    successDecryptionInspector: function Comp(props) {
+        if (!readTypedMessageMetadata(props.message.meta, 'poll', {}).ok) return null
```

Comment:
Please create a metadata reader using `createTypedMessageMetadataReader` instead of calling the underlying function. You can see an example here. https://github.com/DimensionDev/Maskbook/blob/4250dc7f7346175089b2eb4ce32068c8ddd1d370/src/plugins/FileService/define.tsx#L10-L10

The second arg is the JSON Schema (if you have written it).